ogre2/src/Ogre2BoundingBoxCamera.cc

```diff
@@ -0,0 +1,1469 @@
+  {
+    auto ogreIds = nameToOgreIds.second;
+
+    // If not a multi-link model, add the 3d box from the OGRE API
+    if (ogreIds.size() == 1)
+    {
+      auto box = this->dataPtr->boundingboxes[ogreIds[0]];
+      this->dataPtr->outputBoxes.push_back(*box);
+    }
+    else
+    {
+      std::vector<math::Vector3d> vertices;
+
+      // Get all the 3D vertices of the sub-items(total mesh)
+      this->MeshVertices(ogreIds, vertices);
+
+      // Get the oriented bounding box from the mesh using PCA
+      math::OrientedBoxd mergedBox = math::eigen3::verticesToOrientedBox(
+        vertices);
+
+      // convert to rendering::BoundingBox format
+      BoundingBox box;
+      auto pose = mergedBox.Pose();
+      box.Center(pose.Pos());
+      box.Orientation(pose.Rot());
+      box.Size(mergedBox.Size());
+      box.Label(this->dataPtr->visibleBoxesLabel[ogreIds[0]]);
+
+      this->dataPtr->outputBoxes.push_back(box);
+    }
+  }
+
+  // reverse the order of the boxes (useful in testing)
+  std::reverse(this->dataPtr->outputBoxes.begin(),
+    this->dataPtr->outputBoxes.end());
+}
+
+/////////////////////////////////////////////////
+void Ogre2BoundingBoxCamera::MergeMultiLinksModels2D()
+{
+  // Combine the boxes with the same parent name together to merge them
+  for (auto box : this->dataPtr->boundingboxes)
+  {
+    auto ogreId = box.first;
+    auto parentName = this->dataPtr->materialSwitcher->ogreIdName[ogreId];
+    this->dataPtr->parentNameToBoxes[parentName].push_back(box.second);
+  }
+
+  // Merge the boxes that is related to the same parent
+  for (auto nameToBoxes : this->dataPtr->parentNameToBoxes)
+  {
+    auto mergedBox = this->MergeBoxes2D(nameToBoxes.second);
+
+    // Store boxes in the output vector
+    this->dataPtr->outputBoxes.push_back(mergedBox);
+  }
+
+  // reverse the order of the boxes (usful in testing)
+  std::reverse(this->dataPtr->outputBoxes.begin(),
+    this->dataPtr->outputBoxes.end());
+}
+
+/////////////////////////////////////////////////
+BoundingBox Ogre2BoundingBoxCamera::MergeBoxes2D(
+        const std::vector<BoundingBox *> &_boxes)
+{
+  if (_boxes.size() == 1)
+    return *_boxes[0];
+
+  BoundingBox mergedBox;
+  uint32_t minX = UINT32_MAX;
+  uint32_t maxX = 0;
+  uint32_t minY = UINT32_MAX;
+  uint32_t maxY = 0;
+
+  for (auto box : _boxes)
+  {
+    uint32_t boxMinX = box->Center().X() - box->Size().X() / 2;
+    uint32_t boxMaxX = box->Center().X() + box->Size().X() / 2;
+    uint32_t boxMinY = box->Center().Y() - box->Size().Y() / 2;
+    uint32_t boxMaxY = box->Center().Y() + box->Size().Y() / 2;
+
+    minX = std::min<uint32_t>(minX, boxMinX);
+    maxX = std::max<uint32_t>(maxX, boxMaxX);
+    minY = std::min<uint32_t>(minY, boxMinY);
+    maxY = std::max<uint32_t>(maxY, boxMaxY);
+  }
+
+  uint32_t width = maxX - minX;
+  uint32_t height = maxY - minY;
+  mergedBox.Size({width, height, 0});
+  mergedBox.Center({minX + width / 2, minY + height / 2, 0});
```

Comment:
```suggestion
  mergedBox.Size({static_cast<double>(width), static_cast<double>(height), 0});
  
  mergedBox.Center({static_cast<double>(minX) + width / 2,
      static_cast<double>(minY) + height / 2, 0});
```

Prevent narrowing conversion warning.